docs/modeling/bounding-boxes.rst

```diff
@@ -0,0 +1,216 @@
+        ...
+        >>> Sphere3D = custom_model(sphere)
+        >>> Sphere3D.bounding_box_default = sphere_bb
+        >>> model = Sphere3D()
+        >>> model.bounding_box = 'auto'
+        >>> model.bounding_box
+        ((-5.0, 5.0), (-5.0, 5.0), (-5.0, 5.0))
+
+Efficient evaluation with ``render_model``
+------------------------------------------
+
+When a model is evaluated over a range much larger than the model itself, it may 
+be prudent to use `astropy.modeling.render_model` if efficiency is a concern. 
+The ``render_model`` function can be used to evaluate a model on an array of the 
+same dimensions. If no array is given, ``render_model`` will return a "postage 
+stamp" array corresponding to the bounding box region. However, if 
+``bounding_box`` is `None` an image or coordinates must be passed. To update the 
+input array in place, set ``copy = False``.
+
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        import matplotlib.pyplot as plt
+        from astropy.modeling import models, render_model
+
+        imshape = (450, 600)
+        nsrc = 50
+        np.random.seed(0)
+
+        # Generate a random list of models:
+
+        model_params = [dict(amplitude=np.abs(np.random.uniform(0, 10)),
+                        x_mean = np.random.uniform(0, imshape[1]),
+                        y_mean = np.random.uniform(0, imshape[0]),
+                        x_stddev = np.abs(np.random.uniform(3, 6)),
+                        y_stddev = np.abs(np.random.uniform(3, 6)),
+                        theta = np.random.uniform(0, 2 * np.pi)) for i in range(nsrc)]
+
+        model_list = [models.Gaussian2D(**kwargs) for kwargs in model_params]
+
+        # Evaluate models over the full image with and without bounding boxes
+
+        bb_array = np.zeros(imshape)
+        full_array = np.zeros(imshape)
+
+        for model in model_list:
+            render_model(model, bb_array, copy=False)
+            model.bounding_box = None
+            render_model(model, full_array, copy=False)
+
+        max_err = (full_array - bb_array).max()
+
+        plt.figure(figsize=(16, 6))
+        plt.subplots_adjust(wspace=.1, right=1, left=0.1)        
+        plt.subplot(121)
+        plt.imshow(np.sqrt(bb_array), cmap='cubehelix')
+        plt.colorbar()
+        plt.title('Image with Bounded Models', fontsize=16)
+        plt.xlabel('x', fontsize=14)
+        plt.ylabel('y', fontsize=14)
+        plt.subplot(122)
+        plt.imshow(full_array - bb_array, cmap='RdBu', 
+                    vmin=-max_err, vmax=max_err)
+        plt.colorbar(format='%.1e')
+        plt.title('Difference from Full Image, Total Flux Err = %.0e' %
+                ((np.sum(full_array) - np.sum(bb_array)) / np.sum(full_array)),
+                fontsize=16)
+        plt.xlabel('x', fontsize=14)
+        plt.ylabel('y', fontsize=14)
+        plt.show()
+
+At these scales the absolute decrease in evaluation time is small, but the 
```

Comment:
You have two long code examples that are basically the same ... that's not time well-spent for a reader that wants to quickly learn about this.
I'd suggest you combine them into one ... an example where the computation without bounding boxes takes ~ 1 second (to keep the Sphinx build fast) and the computation with bounding boxes is a factor 10 or 100 faster.

You can then still make a comment that the speedup for bounding boxes is larger for larger images and smaller bounding boxes. I don't think the speedup should depend on the number of sources, so that statement should be removed.